Module#04/security-service/src/main/java/com/epam/esm/security/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.web.cors.CorsConfiguration;
+import org.springframework.web.cors.CorsConfigurationSource;
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+
+/**
+ * Configuration class for web security.
+ */
+@Slf4j
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class WebSecurityConfig {
+    private final JwtAuthorizationFilter authorizationFilter;
+    private final AuthenticationProvider authenticationProvider;
+    private final LogoutHandler logoutHandler;
+
+    /**
+     * Configures the security filter chain for HTTP requests.
+     *
+     * @param http The HttpSecurity object to configure.
+     * @return The security filter chain.
+     * @throws Exception if an error occurs during configuration.
+     */
+    @Bean
+    public SecurityFilterChain securityFilterChain(
+            final HttpSecurity http) throws Exception {
+        return http
+                .headers()
+                .frameOptions().sameOrigin()
+                .and()
+                .csrf().disable()
+                .cors().configurationSource(corsConfigurationSource())
+                .and()
+                .authorizeHttpRequests(authorize -> {
+                    try {
+                        authorize.requestMatchers(PathRequest.toH2Console()).hasAuthority("ROLE_ADMIN")
+                                .requestMatchers(POST, "/logout", "/login", "/token/**").permitAll()
+                                .requestMatchers(GET, "/certificates/**").permitAll()
+                                .requestMatchers("/**").hasAuthority("ROLE_ADMIN")
+                                .requestMatchers(POST, "/orders/**").hasAuthority("ROLE_USER")
+                                .anyRequest()
+                                .authenticated()
+                                .and()
+                                .logout()
+                                .logoutUrl("/logout")
+                                .addLogoutHandler(logoutHandler)
+                                .logoutSuccessHandler((request, response, authentication) -> {
+                                    SecurityContextHolder.clearContext();
+                                    response.sendRedirect("/api/token/login");
+                                })
+                                .and()
+                                .formLogin(formLogin -> formLogin.defaultSuccessUrl("/certificates"));
```

Comment:
If I am not wrong formLogin() method is used in MVC with jsp pages or thymleaf, it will redirect to HTML page in stead to redirect to endpoint. In REST there is no  such thing as redirect, this is why HATEOAS is used

Reply:
> If I am not wrong formLogin() method is used in MVC with jsp pages or thymleaf, it will redirect to HTML page in stead to redirect to endpoint. In REST there is no such thing as redirect, this is why HATEOAS is used

> Yes, you are right, this is used to redirect to the login page that the framework generates. This has been removed